src/server.h

```diff
@@ -1000,6 +1049,12 @@ typedef struct client {
      * before adding it the new value. */
     uint64_t client_cron_last_memory_usage;
     int      client_cron_last_memory_type;
+
+    listNode *ref_repl_buf_node; /* Referenced node of replication buffer blocks,
```

Comment:
Actually, i tried to did that, but i gave up. first, replication backlog don't need `ref_block_pos`, and this way make the variable very long. such as, we access `ref_repl_buf_node`, maybe `server.repl_backlog->repl_buf_ref->ref_repl_buf_block`

@oranagra  WDYT？

Reply:
since we don't have generic code that can act on that struct (common to both backlog and replica), i don't think this is a good idea.
even if we could extract a 2 line function that's common, i think doing that (both the extraction of the function, and the extraction of the struct members), would cause more confusion.

i combine this with Yoav's [suggestion](https://github.com/redis/redis/pull/9166#discussion_r714552608) to completely eliminate either the backlog reference or the head link to the buffer chain (a suggestion that i personally don't really like), makes me feel that this suggestion to unify the structs is probably not a good idea.

Reply:
It depends on how do we envision this global replication buffer list will be consumed in the long term. Are replica clients gonna be the only consumers of it? Or do we see this global list as a generic "change data capture stream" that will potentially have more use cases/consumers? If we do, then the redundancy will only grow. E.g. more "consumer" structs will need to define their own `ref_repl_buf_node` and `ref_block_pos` and potentially more metadata that "describes" a pointer/reference into the global list.

Just a thought.

re: `replication backlog don't need ref_block_pos`

Yes, replication backlog's `ref_block_pos` is virtually always `0`, but it can still have it. Doesn't hurt, and may allow us to control replication backlog size in finer granularity(not important though).